texar/modules/encoders/xlnet_encoders.py

```diff
@@ -0,0 +1,606 @@
+
+        "mem_len": int
+            Length of the memory to be used during attention score calculation.
+
+        "reuse_len": int
+            Length of the memory that can be re-used
+
+        "initializer" : dict, optional
+            Hyperparameters of the default initializer that initializes
+            variables created in this module.
+            See :func:`~texar.core.get_initializer` for details.
+
+        "num_heads": int
+            Number of heads in the attention
+
+        "hidden_dim": int
+            Hidden dimension of the embeddings
+
+        "head_dim": int
+            Size of the vectors after head projection.
+
+        "dropout": float
+            Dropout rate for layers
+
+        "attention_dropout": float
+            Dropout rate for attention layers
+
+        "use_segments": bool
+            Boolean to indicate if the input has segments
+
+        "ffn_inner_dim": int
+            Dimension of PositionWise FF network's hidden layer
+
+        "activation": str or callable
+            Activation function applied to the output of the PositionWise FF.
+            See :func:`~texar.core.get_activation_fn` for more details.
+
+        "vocab_size" : int
+            The vocabulary size of `inputs` in `XLNet`.
+
+        "max_seq_len": int
+            Maximum len of the sequence allowed in one segment
+
+        "name" : str
+            Name of the module.
+        """
+
+        return {
+            "name": "xlnet_encoder",
+            'pretrained_model_name': 'xlnet-base-cased',
+            "untie_r": True,
+            "num_layers": 12,
+            "mem_len": 0,
+            "reuse_len": 0,
+            # initializer
+            "initializer": None,
+            # layer
+            "num_heads": 12,
+            "hidden_dim": 768,
+            "head_dim": 64,
+            "dropout": 0.1,
+            "attention_dropout": 0.1,
+            "use_segments": True,
+            # ffn
+            "ffn_inner_dim": 3072,
+            "activation": 'gelu',
+            # embedding
+            "vocab_size": 32000,
+            "max_seq_len": 512,
+            '@no_typecheck': ['pretrained_model_name']
+        }
+
+    def param_groups(self, lr=None, lr_layer_scale=1.0,
+                     decay_base_params=False):
+        r"""Create parameter groups for optimizers. When
+        :attr:`lr_layer_decay_rate` is not 1.0, parameters from each layer form
+        separate groups with different base learning rates.
+
+        This method should be called before applying gradients to the variables
+        through the optimizer. Particularly, after calling the optimizer's
+        `compute_gradients` method, the user can call this method to get
+        variable-specific learning rates for the network. The gradients for each
+        variables can then be scaled accordingly. These scaled gradients are
+        finally applied by calling optimizer's `apply_gradients` method.
```

Comment:
The explanation is good but still a bit difficult to understand. Can we add an example piece of code, like [here](https://github.com/asyml/texar/blob/master/texar/modules/decoders/rnn_decoders.py#L144). The example code should be simple

Reply:
Added an example in b667dd6